web/src/app/schedules/calendar-subscribe/CalendarSubscribeEditDialog.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react'
+import { PropTypes as p } from 'prop-types'
+import gql from 'graphql-tag'
+import { useQuery, useMutation } from '@apollo/react-hooks'
+import FormDialog from '../../dialogs/FormDialog'
+import CalendarSubscribeForm, {
+  reminderMinutesOptions,
+} from './CalendarSubscribeForm'
+import { GenericError, ObjectNotFound } from '../../error-pages'
+import _ from 'lodash-es'
+import Spinner from '../../loading/components/Spinner'
+import { fieldErrors } from '../../util/errutil'
+
+const query = gql`
+  query($id: ID!) {
+    userCalendarSubscription(id: $id) {
+      id
+      name
+      reminderMinutes
+      scheduleID
+    }
+  }
+`
+
+const mutation = gql`
+  mutation($input: UpdateUserCalendarSubscriptionInput!) {
+    updateUserCalendarSubscription(input: $input)
+  }
+`
+
+/*
+ * Load edit data here before rendering edit content to
+ * avoid breaking any rules of hooks
+ */
+export default function CalendarSubscribeEditDialog(props) {
+  const { data, loading, error } = useQuery(query, {
+    variables: { id: props.calSubscriptionID },
+  })
+
+  if (error) return <GenericError error={error.message} />
+  if (!_.get(data, 'userCalendarSubscription.id')) {
+    return loading ? <Spinner /> : <ObjectNotFound />
+  }
+
+  return (
+    <CalendarSubscribeEditDialogContent
+      data={data.userCalendarSubscription}
+      onClose={props.onClose}
+    />
+  )
+}
+
+export function CalendarSubscribeEditDialogContent(props) {
+  const { data, onClose } = props
+
+  // set default values from retrieved data
+  const [value, setValue] = useState({
+    name: _.get(data, 'name', ''),
+    scheduleID: _.get(data, 'scheduleID', null),
+    reminderMinutes: _.get(data, 'reminderMinutes', []).map(r =>
+      reminderMinutesOptions.find(opt => opt.value === r),
+    ),
+  })
+
+  // setup the mutation
+  const [updateSubscription, updateSubscriptionStatus] = useMutation(mutation, {
+    variables: {
+      input: {
+        id: props.data.id,
+        name: value.name,
+        reminderMinutes: value.reminderMinutes.map(r => r && r.value),
+      },
+    },
+    onCompleted: () => props.onClose(),
+  })
+
+  return (
+    <FormDialog
+      title='Edit Calendar Subscription'
+      onClose={onClose}
+      loading={updateSubscriptionStatus.loading}
+      onSubmit={() => updateSubscription()}
+      form={(
+        <CalendarSubscribeForm
+          disableSchedField
+          errors={fieldErrors(updateSubscriptionStatus.error)}
+          loading={updateSubscriptionStatus.loading}
+          onChange={setValue}
+          value={value}
+        />
+      )}
+    />
+  )
+}
+
+CalendarSubscribeEditDialog.propTypes = {
```

Comment:
@mastercactapus 

- **Maybe use "Reminder" or "Notification" instead of "Alarm" for reminderMinutes? I know at least on Android, Alarm means something different than calendar event notifications**
  - Valid, I'll run with `Reminder` as it works well as a noun while in both singular and plural forms imo (i.e. "Reminder, Second Reminder, etc" as opposed to "Notification, Second Notification, etc")
- **No need to require first alarm, they can change it later if they want and the template will still render correctly without it**
  - Sounds good. I'm trying to think of this from a user's perspective tho– if first reminder is optional and they submit without setting it, what would they expect the initial reminder duration to be, and how would we signify that to the user? Some questions we should ask ourselves before adjusting
- **Seems a little odd to have a second Subscribe button that does something different than the first (in completed dialog), no immediate ideas though**
  - Yeah, I was also going back and forth on this. I was thinking of either changing the text, or adding more text to the success form subheader. I eventually landed on the icon that signifies being opened in a new tab, but you're right that it could be better. No immediate ideas myself either. @m17ch @arurao @dctalbot @KatieMSB?
- **extra space with 2+: "You have 2 active subscription s for this schedule"**
  - 👍
- **Edit dialog: schedule should be read-only**
  - 👍